src/components/views/dialogs/DevtoolsDialog.js

```diff
@@ -0,0 +1,294 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import sdk from 'matrix-react-sdk';
+import { _t } from 'matrix-react-sdk/lib/languageHandler';
+import MatrixClientPeg from 'matrix-react-sdk/lib/MatrixClientPeg';
+
+class SendCustomEvent extends React.Component {
+    static propTypes = {
+        roomId: React.PropTypes.string.isRequired,
+        onBack: React.PropTypes.func.isRequired,
+    };
+
+    constructor(props, context) {
+        super(props, context);
+        this._send = this._send.bind(this);
+        this.onBack = this.onBack.bind(this);
+    }
+
+    state = {
+        message: null,
+    };
+
+    onBack() {
+        if (this.state.message) {
+            this.setState({ message: null });
+        } else {
+            this.props.onBack();
+        }
+    }
+
+    _buttons() {
+        return <div className="mx_Dialog_buttons">
+            <button onClick={this.onBack}>{ _t('Back') }</button>
+            {!this.state.message && <button onClick={this._send}>{ _t('Send') }</button>}
+        </div>;
+    }
+
+    send(content) {
+        return MatrixClientPeg.get().sendEvent(this.props.roomId, this.refs.eventType.value, content);
+    }
+
+    async _send() {
+        let message;
+        try {
+            const content = JSON.parse(this.refs.evContent.value);
+            await this.send(content);
+            message = _t('Event sent!');
+        } catch (e) {
+            message = `${_t('Failed to send custom event.')} (${e.toString()})`;
+        }
+        this.setState({ message });
+    }
+
+    render() {
+        if (this.state.message) {
+            return <div>
+                <div className="mx_Dialog_content">
+                    {this.state.message}
+                </div>
+                {this._buttons()}
+            </div>;
+        }
+
+        return <div>
+            <div className="mx_Dialog_content">
+                <div className="mx_TextInputDialog_label">
+                    <label htmlFor="eventType"> { _t('Event Type') } </label>
+                </div>
+                <div>
+                    <input id="eventType" ref="eventType" className="mx_TextInputDialog_input" size="64" />
+                </div>
+
+                <div className="mx_TextInputDialog_label">
+                    <label htmlFor="evContent"> { _t('Event Content') } </label>
+                </div>
+                <div>
+                    <textarea id="evContent" ref="evContent" className="mx_TextInputDialog_input" defaultValue={"{\n\n}"} cols="63" rows="5" />
+                </div>
+            </div>
+            {this._buttons()}
+        </div>;
+    }
+}
+
+class SendCustomStateEvent extends SendCustomEvent {
+    send(content) {
+        return MatrixClientPeg.get().sendStateEvent(this.props.roomId, this.refs.eventType.value, content,
+            this.refs.stateKey.value);
+    }
+
+    render() {
```

Comment:
Given the only difference between this `render` and the `render` being clobbered, I think you should have `this._content()` that can be clobbered to return the correct form for the given component. This keeps the `render` the same.